src/components/number-input/number-input.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import AddIcon from '@material-ui/icons/Add';
+import RemoveIcon from '@material-ui/icons/Remove';
+import Button from '@material-ui/core/Button';
+
+import { useStyles } from './number-input.styles';
+
+const NumberInput = ({ onChangeQuantity, quantity }) => {
+  const styles = useStyles();
+
+  return (
+    <div className={styles.root}>
+      <Button
+        className={styles.button}
+        onClick={() => onChangeQuantity(-1)}
+        disabled={quantity <= 1}
+      >
+        <RemoveIcon />
+      </Button>
+      <input
+        className={styles.input}
+        type='number'
+        value={quantity}
+        onChange={(e) => onChangeQuantity(e.target.value, true)}
```

Comment:
Why do you pass to the `onChangeQuantity` as a second argument `true` if in `CartItem` this function takes as the second argument a key? I don't think that `true` is an appropriate value for `key`. Do I miss something?

Reply:
Yes, this is a way to distinguish user actions - press +/- or enter input.
Line 63, cart.sagas.js
` item.quantity = key ? value || 1 : item.quantity + value;`